bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/values/BArray.java

```diff
@@ -208,4 +209,10 @@
     BArray slice(long startIndex, long endIndex);
 
     void setLength(long i);
+
+    /**
+     * Check and and set the array type as readonly.
+     * @param type original Array type
+     */
+    BArray checkAndSetReadonly(ArrayType type);
```

Comment:
Is this change needed? The inherent type of a value cannot change once the value is created, so IMO, we shouldn't have such a method (which allows changing the type) in the public API.

If your requirement is to create an immutable value, IMO it should be done by allowing the user to pass both the type and the values and internally calling the relevant constructors that accept the same;
- [io.ballerina.runtime.values.ArrayValueImpl#ArrayValueImpl(io.ballerina.runtime.api.types.ArrayType, long, io.ballerina.runtime.values.ListInitialValueEntry[])](https://github.com/ballerina-platform/ballerina-lang/blob/v2.0.0-Preview5/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/values/ArrayValueImpl.java#L187) 
- [io.ballerina.runtime.values.MapValueImpl#MapValueImpl(io.ballerina.runtime.api.types.Type, io.ballerina.runtime.api.values.BMapInitialValueEntry[])](https://github.com/ballerina-platform/ballerina-lang/blob/v2.0.0-Preview5/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/values/MapValueImpl.java#L106)

Reply:
With the clarification from spec issue #646 we wouldn't need this API.